ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -941,16 +940,20 @@ public:
 #endif
 
 #ifndef DNSDIST
-  void del()
+  void del(const MDBInVal& key)
   {
-    MDBOutVal key, val;
-
     if (LMDBLS::s_flag_deleted) {
-      int rc_get = mdb_cursor_get (*this, &key.d_mdbval, &val.d_mdbval, MDB_GET_CURRENT);
+#ifdef PARANOIA
+      MDBOutVal key2{};
+      MDBOutVal val{};
+      int rc_get = mdb_cursor_get (*this, &key2.d_mdbval, &val.d_mdbval, MDB_GET_CURRENT);
 
-      if(rc_get) {
+      // XXX what to do if MDB_NOTFOUND?
+      if (rc_get) {
         throw std::runtime_error("getting key to mark data as deleted: " + std::string(mdb_strerror(rc_get)));
       }
+      // XXX enforce check that key == key2?
```

Comment:
that -should- only happen when we have a bug, right? and the _get is not free (not expensive either). I'm unsure we should leave the PARANOIA block in at all, but it might be good to use it during tests until we are sure the magical dis/reappearing records are solved.

Reply:
Plus the classic: you might be paranoid, but that does not mean they're not after you.